tests/fixtures/new_fixtures.py

```diff
@@ -33,3 +35,12 @@ def empty_logger(minimal_state: State) -> Logger:
 @pytest.fixture(autouse=True)
 def disable_wandb(monkeypatch: pytest.MonkeyPatch):
     monkeypatch.setenv("WANDB_MODE", "disabled")
+
+
+# Class-scoped temporary directory. That deletes itself. This is useful for e.g. not
+# writing too many checkpoints.
+@pytest.fixture(scope='class')
+def self_destructing_tmp(tmp_path_factory: pytest.TempPathFactory):
+    my_tmpdir = tmp_path_factory.mktemp("checkpoints")
+    yield my_tmpdir
+    shutil.rmtree(str(my_tmpdir))
```

Comment:
If the test fails, do we lose the tempdir? (It is helpful to keep it around for debugging a failing test)